common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1554,8 +1554,23 @@ def _get_raw_parent_location(self, location, revision=ModuleStoreEnum.RevisionOp
             or revision == ModuleStoreEnum.RevisionOption.draft_preferred
 
         parent_cache = self._get_parent_cache(self.get_branch_setting())
+
+        def has_path_to_root(xblock):
```

Comment:
Is this expensive? I'm less worried about fixing this in old mongo, but if this isn't expensive to implement, great :)